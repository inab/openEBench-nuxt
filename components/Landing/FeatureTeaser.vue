<template>
	<v-container>
		<h2 class="text-h4 mb-15">What can you do in OpenEBench?</h2>
		<v-row>
			<v-col align="center" cols="12" md="3">
				<feature-card to="benchmarking">
					<template #icon>
						<v-icon dark>$vuetify.icons.values.benchmark</v-icon>
					</template>
					<template #title> Benchmarks </template>
					<template #description
						>Explore existing community-driven scientific benchmark results for
						Life Sciences software</template
					>
				</feature-card>
			</v-col>
			<v-col align="center" cols="12" md="3" class="mt-10 mt-md-0">
				<feature-card to="tool">
					<template #icon>
						<v-icon large dark>mdi-clipboard-list-outline</v-icon>
					</template>
					<template #title> Tools & Metrics </template>
					<template #description
						>Individually browse technical metrics for popular research software
						in Life Sciences</template
					>
				</feature-card>
			</v-col>
			<v-col align="center" cols="12" md="3" class="mt-10 mt-md-0">
				<feature-card to="observatory">
					<template #icon>
						<v-icon dark>$vuetify.icons.values.observatory</v-icon>
					</template>
					<template #title> Tools Observatory </template>
					<template #description>
						Quantitative monitoring of the technical quality of software in Life
						Sciences</template
					>
				</feature-card>
			</v-col>
			<v-col align="center" cols="12" md="3" class="mt-10 mt-md-0">
				<feature-card to="projects">
					<template #icon>
						<v-icon dark>$vuetify.icons.values.projects</v-icon>
					</template>
					<template #title> Project Spaces </template>
					<template #description>
						Community spaces for projects to collaborate on software in life
						sciences</template
					>
				</feature-card>
			</v-col>
		</v-row>
	</v-container>
</template>

<script>
import FeatureCard from '~/components/Cards/FeatureCard';

export default {
	name: 'FeatureTeaser',
	components: { FeatureCard },
};
</script>
