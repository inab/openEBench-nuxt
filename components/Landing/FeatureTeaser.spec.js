import { mount } from '@vue/test-utils';
import FeatureTeaser from './FeatureTeaser.vue';

const factory = () => {
	return mount(FeatureTeaser, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
	});
};

describe('FeatureTeaser', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});
});
