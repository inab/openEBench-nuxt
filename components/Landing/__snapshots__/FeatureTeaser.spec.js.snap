// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureTeaser should match snapshot 1`] = `
"<div class=\\"container\\">
  <h2 class=\\"text-h4 mb-15\\">What can you do in OpenEBench?</h2>
  <div class=\\"row\\">
    <div align=\\"center\\" class=\\"col-md-3 col-12\\">
      <div>
        <div class=\\"v-avatar mb-5 mb-md-10 primary\\" style=\\"height: 62px; min-width: 62px; width: 62px;\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\">$vuetify.icons.values.benchmark</i></div>
        <h2 class=\\"text-h6 mb-2\\"> Benchmarks </h2>
        <p class=\\"body-1\\">Explore existing community-driven scientific benchmark results for
          Life Sciences software</p> <a class=\\"mt-2 v-btn v-btn--outlined v-btn--router theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Explore </span></a>
      </div>
    </div>
    <div align=\\"center\\" class=\\"mt-10 mt-md-0 col-md-3 col-12\\">
      <div>
        <div class=\\"v-avatar mb-5 mb-md-10 primary\\" style=\\"height: 62px; min-width: 62px; width: 62px;\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-clipboard-list-outline theme--dark\\" style=\\"font-size: 36px;\\"></i></div>
        <h2 class=\\"text-h6 mb-2\\"> Tools &amp; Metrics </h2>
        <p class=\\"body-1\\">Individually browse technical metrics for popular research software
          in Life Sciences</p> <a class=\\"mt-2 v-btn v-btn--outlined v-btn--router theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Explore </span></a>
      </div>
    </div>
    <div align=\\"center\\" class=\\"mt-10 mt-md-0 col-md-3 col-12\\">
      <div>
        <div class=\\"v-avatar mb-5 mb-md-10 primary\\" style=\\"height: 62px; min-width: 62px; width: 62px;\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\">$vuetify.icons.values.observatory</i></div>
        <h2 class=\\"text-h6 mb-2\\"> Tools Observatory </h2>
        <p class=\\"body-1\\">
          Quantitative monitoring of the technical quality of software in Life
          Sciences</p> <a class=\\"mt-2 v-btn v-btn--outlined v-btn--router theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Explore </span></a>
      </div>
    </div>
    <div align=\\"center\\" class=\\"mt-10 mt-md-0 col-md-3 col-12\\">
      <div>
        <div class=\\"v-avatar mb-5 mb-md-10 primary\\" style=\\"height: 62px; min-width: 62px; width: 62px;\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\">$vuetify.icons.values.projects</i></div>
        <h2 class=\\"text-h6 mb-2\\"> Project Spaces </h2>
        <p class=\\"body-1\\">
          Community spaces for projects to collaborate on software in life
          sciences</p> <a class=\\"mt-2 v-btn v-btn--outlined v-btn--router theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Explore </span></a>
      </div>
    </div>
  </div>
</div>"
`;
